benzina/src/json/nullable.rs

```diff
@@ -0,0 +1,150 @@
+use std::fmt::Debug;
+
+use diesel::{
+    deserialize::{FromSql, FromSqlRow},
+    expression::AsExpression,
+    pg::{Pg, PgValue},
+    serialize::{IsNull, ToSql},
+    sql_types::Nullable,
+};
+use serde::{Serialize, de::DeserializeOwned};
+
+macro_rules! impl_nullable {
+    (
+        $($type:ident => $diesel_type:ident => $uppercase_diesel_type:ident => $serializer:path => $deserializer:path => $diesel_type_import:path),*
+    ) => {
+        $(
+            #[doc = concat!("A diesel [`Nullable`]<[`", stringify!($diesel_type), "`]> serialization and")]
+            #[doc = "deserialization wrapper"]
+            #[doc = ""]
+            #[doc = ""]
+            #[doc = concat!("For a `nullable` ", stringify!($uppercase_diesel_type), " column in diesel, you would use the [`serde_json::Value`] model field.<br>")]
+            #[doc = concat!("If you need the control over the ", stringify!($uppercase_diesel_type), " structure you need to create your custom type that implements [`FromSql`] and [`ToSql`].")]
+            #[doc = ""]
+            #[doc = concat!("This wrapper allows to control the ", stringify!($uppercase_diesel_type), " structure without reimplementing that traits for each ", stringify!($uppercase_diesel_type), " column.")]
+            #[doc = ""]
+            #[doc = "This type is not intended to be used directly in the model but rather to be used in with diesel [`serialize_as`] and [`deserialize_as`]."]
+            #[doc = "```"]
+            #[doc = concat!("use benzina::{", stringify!($type), "};")]
+            #[doc = "use diesel::{Queryable, Insertable, sql_types::Nullable};"]
+            #[doc = ""]
+            #[doc = "#[derive(Debug, Queryable)]"]
+            #[doc = "#[diesel(table_name = users)]"]
+            #[doc = "struct User {"]
+            #[doc = "    id: u32,"]
```

Comment:
This wouldn't really work otherwise.

```suggestion
            #[doc = "    id: U31,"]
```